### ISSUE TITLE
An option to save the output of `dvc run` despite an unclean termination.

### ISSUE THREAD

**Issue by yukw777** [iterative/dvc#1176]
I've been using dvc to train some deep learning networks that take days to train. As a result, I don't typically run my training script to completion; rather, I closely monitor the training loss, and terminate the script (with Ctrl+c) when the model starts to get overtrained. Unfortunately, this means `dvc run` will not save the models I saved until then. 

I'm wondering if it'd be a good idea to add an option to `dvc run` that would allow saving the output even though the command may not have finished running (be it Ctrl+c or error).

I don't think I've mentioned it in all the issues I submitted, but I'd like to thank you for an awesome project. It's made my workflow so much easier, and things are still improving so fast! I can't wait to see what the future holds for dvc.

**Comment by efiop** [comment]
Hi @yukw777 !

That is a very interesting scenario you have there! But I don't think this should be solved by the dvc itself, since it seems pretty unnatural for a stage in a pipeline to be saved when it is interrupted like so. Would it be possible to configure a condition in your script for when it should stop training so you could set it when starting a script? Or, if you really want to keep the interactivity, would it be possible to modify your script so it polls for a key press(e.g. ESC or ENTER) and stops training when it is detected? I think it would make the termination much less violent and seem less of an afterthought.

Thanks,
Ruslan

**Comment by yukw777** [comment]
That's actually a better solution. Thank you for entertaining my idea!
